File: backend/FitHub/src/main/java/com/uche/fithub/dto/customer_dto/AddCustomerSchema.java
package com.uche.fithub.dto.customer_dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Data;

@Data
public class AddCustomerSchema {

    @NotBlank(message = "Champ obligatoire")
    private String firstName;
    @NotBlank(message = "Champ obligatoire")
    private String lastName;

    @Pattern(regexp = "^\\+?[0-9]{8}$", message = "Le numéro doit comporter 8 chiffres")
    private String phoneNumber;

}

File: frontend/fithub_front/src/app/components/customer/models/AddCustomerSchema.ts
export class AddCustomerSchema {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  constructor(
    firstName: string = '',
    lastName: string = '',
    phoneNumber: string = '',
  ) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.phoneNumber = phoneNumber;
  }
}
